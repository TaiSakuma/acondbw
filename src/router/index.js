import Vue from "vue";
import VueRouter from "vue-router";
import Home from "@/views/Home.vue";
import ProductTop from "@/views/ProductTop.vue";

import ALL_MAPS from "@/graphql/AllMaps.gql";

Vue.use(VueRouter);

const routes = [
  {
    path: "/",
    name: "home",
    component: Home,
  },
  {
    path: "/about",
    name: "about",
    component: () =>
      import(/* webpackChunkName: "about" */ "@/views/About.vue"),
  },
  {
    path: "/simulations",
    component: ProductTop,
    props: {
      title: "Simulations",
      icon: "mdi-creation",
      itemPageName: "SimulationItem",
    },
    children: [
      {
        path: "",
        name: "SimulationList",
        component: () =>
          import(
            /* webpackChunkName: "simulations" */ "@/views/SimulationList.vue"
          ),
      },
      {
        path: "item/:name",
        name: "SimulationItem",
        component: () =>
          import(
            /* webpackChunkName: "simulations" */ "@/views/SimulationItem.vue"
          ),
      },
    ],
  },
  {
    path: "/maps",
    component: ProductTop,
    props: { title: "Maps", icon: "map", itemPageName: "MapItem" },
    children: [
      {
        path: "",
        name: "MapList",
        component: () =>
          import(/* webpackChunkName: "maps" */ "@/views/MapList.vue"),
        props: {
          productTypeName: { singular: "map", plural: "maps" },
          query: ALL_MAPS,
          queryName: "allMaps",
          productIdFieldName: "mapId",
          productItemCard: "MapItemCard",
          productAddForm: "MapAddForm",
          disableAdd: process.env.VUE_APP_ACONDBW_MAP_MUTATION_DIALOG != "true",
        },
      },
      {
        path: "item/:name",
        name: "MapItem",
        component: () =>
          import(/* webpackChunkName: "maps" */ "@/views/MapItem.vue"),
      },
    ],
  },
  {
    path: "/beams",
    component: ProductTop,
    props: {
      title: "Beams",
      icon: "mdi-spotlight-beam",
      itemPageName: "BeamItem",
    },
    children: [
      {
        path: "",
        name: "BeamList",
        component: () =>
          import(/* webpackChunkName: "beams" */ "@/views/BeamList.vue"),
      },
      {
        path: "item/:name",
        name: "BeamItem",
        component: () =>
          import(/* webpackChunkName: "beams" */ "@/views/BeamItem.vue"),
      },
    ],
  },
];

const router = new VueRouter({
  mode: "history",
  base: process.env.BASE_URL,
  routes,
});

export default router;
