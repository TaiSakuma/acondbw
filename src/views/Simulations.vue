<template>
  <div class="simulations">
    <v-container fluid>
      <v-row class="display-1 mx-1 mt-3 primary--text" style="max-width: 980px;">
        <v-col col="8" class="pa-0 ma-0">
          <span class="me-2">
            <v-icon>mdi-creation</v-icon>
          </span>Simulations
          <span v-if="$route.name == 'SimulationItem'">
            <v-icon large color="primary">mdi-chevron-right</v-icon>
            {{ $route.params.name }}
          </span>
        </v-col>
      </v-row>
      <transition :name="transitionName" :mode="transitionMode">
        <keep-alive>
          <router-view :key="$route.fullPath"></router-view>
        </keep-alive>
      </transition>
    </v-container>
  </div>
</template>

<script>
export default {
  name: "Simulations",
  data: () => ({
    transitionName: "fade-simulations-leave",
    transitionMode: "out-in"
  }),
  beforeRouteUpdate(to, from, next) {
    this.transitionName = "fade-simulations-update";
    this.transitionMode = "out-in";
    next();
  },
  beforeRouteLeave(to, from, next) {
    this.transitionName = "fade-simulations-leave";
    this.transitionMode = "out-in";
    next();
  }
};
</script>

<style scoped>
.fade-simulations-update-enter-active {
  transition: opacity 0.2s;
}

.fade-simulations-update-leave-active {
  transition: opacity 0s;
}

.fade-simulations-update-enter,
.fade-simulations-update-leave-to {
  opacity: 0;
}

.fade-simulations-leave-enter-active {
  transition: opacity 0s;
}

.fade-simulations-leave-leave-active {
  transition: opacity 0s;
}

.fade-simulations-leave-enter {
  opacity: 1;
}

.fade-simulations-leave-leave-to {
  opacity: 1;
}
</style>
