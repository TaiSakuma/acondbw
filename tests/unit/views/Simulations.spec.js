
import Vue from "vue";
import VueRouter from "vue-router";
import Vuetify from "vuetify";
import { mount, shallowMount, createLocalVue } from "@vue/test-utils";

import Simulations from "@/views/Simulations.vue";

import router from "@/router";

Vue.use(Vuetify);
Vue.use(VueRouter);

describe("Simulations.vue", () => {
  let localVue;
  let vuetify;

  function createWrapper(loading = false) {
    return shallowMount(Simulations, {
      localVue,
      vuetify,
      router,
      stubs: ["router-link", "router-view"]
    });
  }

  beforeEach(function() {
    localVue = createLocalVue();
    vuetify = new Vuetify();
  });

  it("match snapshot list", async () => {
    const wrapper = createWrapper();
    await router.push({ name: "SimulationList"});
    await Vue.nextTick();
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("match snapshot item", async () => {
    const wrapper = createWrapper();
    await router.push({ name: "SimulationItem", params: { name: "simulation001" } });
    await Vue.nextTick();
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("transition update", async () => {
    const wrapper = createWrapper();
    await router.push({ name: "SimulationList"});
    await router.push({ name: "SimulationItem", params: { name: "simulation001" } });

    // Not clear how to test
    // Neither beforeRouteUpdate() or beforeRouteLeave() is called
    // in the test

    // const trans_attrs = wrapper.find('transition-stub').attributes();
    // expect(trans_attrs.name).toBe('fade-simulations-update');
    // expect(trans_attrs.mode).toBe('out-in');
  });

  it("transition leave", async () => {
    const wrapper = createWrapper();
    await router.push({ name: "SimulationList"});
    await router.push("/about");

    // Not clear how to test
    // Neither beforeRouteUpdate() or beforeRouteLeave() is called
    // in the test

    // const trans_attrs = wrapper.find('transition-stub').attributes();
    // expect(trans_attrs.name).toBe('fade-simulations-leave');
    // expect(trans_attrs.mode).toBe('out-in');
  });

});
