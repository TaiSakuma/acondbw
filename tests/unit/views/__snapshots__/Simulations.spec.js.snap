// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Simulations.vue match snapshot item 1`] = `
<div class="simulations">
  <v-container-stub tag="div" fluid="true">
    <v-row-stub tag="div" class="display-1 mx-1 mt-3 primary--text">
      <v-col-stub tag="div" col="8" class="pa-0 ma-0"><span class="me-2"><v-icon-stub>mdi-creation</v-icon-stub></span>Simulations
        <span><v-icon-stub large="" color="primary">mdi-chevron-right</v-icon-stub>
          simulation001
        </span></v-col-stub>
    </v-row-stub>
    <transition-stub name="fade-simulations-leave" mode="out-in">
      <keep-alive-stub>
        <router-view-stub></router-view-stub>
      </keep-alive-stub>
    </transition-stub>
  </v-container-stub>
</div>
`;

exports[`Simulations.vue match snapshot list 1`] = `
<div class="simulations">
  <v-container-stub tag="div" fluid="true">
    <v-row-stub tag="div" class="display-1 mx-1 mt-3 primary--text">
      <v-col-stub tag="div" col="8" class="pa-0 ma-0"><span class="me-2"><v-icon-stub>mdi-creation</v-icon-stub></span>Simulations
        <!---->
      </v-col-stub>
    </v-row-stub>
    <transition-stub name="fade-simulations-leave" mode="out-in">
      <keep-alive-stub>
        <router-view-stub></router-view-stub>
      </keep-alive-stub>
    </transition-stub>
  </v-container-stub>
</div>
`;
